tools/artifact_gen/artifact_gen.sh

```diff
@@ -28,7 +28,7 @@ tools/bazel query --noimplicit_deps --output=xml 'kind(http_archive, //external:
 
 # PHASE 1: generate artifacts
 cd tools/artifact_gen
-bazel build -c opt :artifact_gen 
+../../tools/bazel build -c opt :artifact_gen 
```

Comment:
`tools/bazel` should be enough? given that you already cd above to the project root?